### ISSUE TITLE
Use `println!` in rust will make the execution terminated on the Windows environment.

### ISSUE THREAD

**Issue by HKMV** [WasmEdge/WasmEdge#3267]
### Summary

我编写了一个服务程序，当我使用println!打印时，服务无法正常运行，代码如下：
``` rust
use std::net::SocketAddr;
use warp::{Filter};

#[tokio::main(flavor = "current_thread")]
async fn main() {
    // GET /
    let help = warp::get()
        .and(warp::path::end())
        .map(|| "Try POSTing data to /echo such as: `curl localhost:8080/echo -XPOST -d 'hello world'`\n");

    // POST /echo
    let echo = warp::post()
        .and(warp::path("echo"))
        .and(warp::body::bytes())
        .map(|body_bytes: bytes::Bytes| {
            format!("{}\n", std::str::from_utf8(body_bytes.as_ref()).unwrap())
        });

    let routes = help.or(echo);

    let addr = SocketAddr::from(([0, 0, 0, 0], 8080));
    // println!("Listening on http://{}", addr); //这里会导致服务无法正常运行
    warp::serve(routes).run(addr).await
}
```
我的系统版本是：Windows11
WasmEdge版本是：0.13.5

后来在WSL(Ubuntu22)中测试是可以正常运行的

### Current State

使用println!会导致程序无故终止。

### Expected State

在使用println!时程序可以正常运行

### Reproduction steps

1. Build
cargo build --target wasm32-wasi --release
2. Execute 
wasmedge wasmedge_hyper_server.wasm
3. Get error
没有错误提示，在输出println!内容后程序就停止了

### Screenshots

![image](https://github.com/WasmEdge/WasmEdge/assets/26456469/1bf909f6-57ae-468b-a1aa-6fd7dd0346ce)

### Components

CLI

### WasmEdge Version or Commit you used

0.13.5

### Operating system information

Windows11

### Hardware Architecture

amd64

**Comment by hydai** [comment]
Thanks for reporting this. I added the `help wanted` tag for those who would like to contribute to this project.

**Comment by TBBle** [comment]
It took me a little while to get it going due to lack of cargo.toml or deps-list, but once I realised this was using the wasmedge forks of tokio and warp, but renamed, I got it working.

```pwsh
cargo add tokio_wasi -F sync,macros,io-util,rt,time --rename tokio
cargo add warp_wasi --rename warp
cargo add bytes
cargo build --target wasm32-wasip1
$Env:CARGO_TARGET_WASM32_WASIP1_RUNNER="wasmedge"
cargo run --target wasm32-wasip1 --release
```

and it seems to work for me, both default and release profiles.

I'm using wasmedge `C:\Program Files\WasmEdge\bin\wasmedge.exe version 0.14.1` and rustc `rustc 1.81.0 (eeb90cda1 2024-09-04)`.

So possibly it has been fixed in wasmedge 0.13.5, or some other dependency has been fixed since March.
